aten/src/ATen/native/mps/operations/TensorCompare.mm

```diff
@@ -416,5 +416,105 @@ Tensor where_mps(const Tensor& condition,
 
 }
 
+Tensor& nan_to_num_out_mps(const Tensor& self,
+                           c10::optional<double> nan,
+                           c10::optional<double> pos_inf,
+                           c10::optional<double> neg_inf,
+                           Tensor& result)
+{
+  TORCH_CHECK(self.scalar_type() == result.scalar_type(), "nan_to_num: dtype of out: ",
+              result.scalar_type(), " should be same as input: ", self.scalar_type());
+  if(result.numel() == 0) {
```

Comment:
Nit (formatting)
Plan to start enforcing clang-format on all .mm files soon to avoid manual edits
```suggestion
  if (result.numel() == 0) {
```